src/foam/nanos/auth/ruler/PreventDuplicateEmailAction.js

```diff
@@ -63,7 +64,7 @@ foam.CLASS({
             )).limit(1).select(count);
 
         if ( count.getValue() == 1 ) {
-          throw new RuntimeException(DUPLICATE_ERROR + user.getEmail());
+          throw new ValidationException(DUPLICATE_ERROR + user.getEmail());
```

Comment:
Can be another ticket, but since this check is occurring on user signup, the translation of messages has to occur client side based on browser settings.
So if we make exceptions DuplicateEmailException and DuplicateUsernameExceptions then we can translate the exception name and any 'exceptionMessage' text we want to make part of the exception.